src/test/ui/issues/issue-52496.rs

```diff
@@ -0,0 +1,14 @@
+struct Foo { bar: f64, baz: i64, bat: i64 }
+
+fn main() {
+    let _ = Foo { bar: .5, baz: 42 };
+    //~^ ERROR numeric float literals must have a significant
```

Comment:
Where does this terminology come from? The error message does not make sense to me.

Reply:
Agreed, the term "significant" (or rather, "significand") has a specific meaning in the realm of floating-point encoding which seems misleading here. I suggest using something like "integer part" instead. The "numeric" can also be dropped since floats can't really be non-numeric (unless you count hexadecimal float literals, which aren't supported anyway).

Reply:
Yeah, I was having trouble coming up with a good message for this. What about "float literals must have a integer part"?

Reply:
*an* integer part, but yeah that sounds good.

Reply:
done